site-scrapers/LowellGeneral/functions.js

```diff
@@ -0,0 +1,212 @@
+
+                const LIs = ul.querySelectorAll("li");
+                let countNotHiddenLIs = 0;
+
+                // There are other LIs that do not represent actual slots
+                // so we don't count those.
+                LIs.map((li) => {
+                    if (
+                        li.rawAttrs.indexOf("hidden") === -1 &&
+                        li.rawAttrs.indexOf("slot") > -1 &&
+                        (li.rawText.includes("am") || li.rawText.includes("pm"))
+                    ) {
+                        countNotHiddenLIs += 1;
+                    }
+                });
+
+                if (countNotHiddenLIs) {
+                    // Ex headerText: Sat  Mar 20
+                    // There's an extra space between the Day of the week
+                    // so we remove double spaces
+                    const headerText = headers.childNodes[
+                        tdIndex
+                    ].innerText.replace(/ {2,}/g, " ");
+                    const thMonth = headerText.split(" ")[1];
+                    const thDay = headerText.split(" ")[2];
+                    let [month, year] = getYearForMonth(thMonth);
+                    const formattedDate = moment([year, month, thDay])
+                        .local()
+                        .format("MM/DD/YYYY");
+                    // length of LIs found is number of appointments
+                    availability[formattedDate] = countNotHiddenLIs;
+                    hasAvailability = true;
+                }
+            }
+        });
+    }
+    return { hasAvailability, availability };
+}
+
+// returns "foobar" in passed node <div id="foobar">
+async function getNodeId(node) {
+    return await node.evaluate((node) => node.id);
+}
+
+// Running fetch() inside the browser page allows use of all the
+// available cookies in an XHR request without needing to build
+// an https request in the node env.
+// returns the XHR response body
+async function getStationTimeslots(page, url) {
+    return await page.evaluate(async (url) => {
+        const response = await fetch(url);
+        return await response.text();
+    }, url);
+}
+
+// This is for testing locally when there are no appointments
+async function getStationTimeslotsWithHttps(url) {
+    return new Promise((resolve) => {
+        https.get(url, (res) => {
+            let body = "";
+            res.on("data", (chunk) => (body += chunk));
+            res.on("end", () => {
+                resolve(body);
+            });
+        });
+    });
+}
+
+// Since the year is not available to us we need to calculate the year.
+// Using the current year is straight forward but the problem is
+// what happens when the month goes from Dec of the current year to Jan of
+// the next year. We add 1 to the current year if the passed month (ex. 'Jan')
+// is less than the current month. Ex. Jan 2022 < December 2021
+function getYearForMonth(passedMonthName) {
```

Comment:
Nice, we can reuse this logic in other places in our codebase (not your responsibility). 
https://github.com/livgust/covid-vaccine-scrapers/blob/master/lib/RxTouch.js#L22